Monika After Story/game/special-effects.rpy

```diff
@@ -436,52 +490,124 @@ init -500 python in mas_parallax:
             half_screen_width = screen_width / 2.0
             half_screen_height = screen_height / 2.0
 
-            # normalize the mouse position with the center of the screen
+            # Normalize the mouse position with the center of the screen
             # 0.0 - left / bottom
             # 1.0 - center
             # 2.0 - right / top
             mouse_x_factor = self.mouse_x / half_screen_width
             mouse_y_factor = self.mouse_y / half_screen_height
 
+            # Offsets from the container
             container_offset_x = abs(self._container.offsets[0])
             container_offset_y = abs(self._container.offsets[1])
 
-            # Our offsets consist of 4 parts:
+            # Offsets from the transform. That's so if the sprite has a custom func
+            # for its transform which modifies these params, we don't have to
+            # account for zoom level in the func, instead we do it here for convenience.
+            tf_x_pos, tf_y_pos = self._transform.pos
+
+            if tf_x_pos is None:
+                tf_x_pos = 0
+
+            elif isinstance(tf_x_pos, float):
+                tf_x_pos = screen_width * tf_x_pos
+
+            if tf_y_pos is None:
+                tf_y_pos = 0
+
+            elif isinstance(tf_y_pos, float):
+                tf_y_pos = screen_height * tf_y_pos
+
+            tf_x_anchor, tf_y_anchor = self._transform.anchor
+            tf_child_x_size, tf_child_y_size = self._transform.child_size
+            tf_render_x_size, tf_render_y_size = self._transform.render_size
+
+            if tf_x_anchor is None:
+                tf_x_anchor = 0
+
+            elif isinstance(tf_x_anchor, float):
+                tf_x_anchor = (tf_x_anchor - tf_render_x_size / 2.0 + tf_child_x_size / 2.0) / tf_child_x_size
+
+            if tf_y_anchor is None:
+                tf_y_anchor = 0
+
+            elif isinstance(tf_y_anchor, float):
+                tf_y_anchor = (tf_y_anchor - tf_render_y_size / 2.0 + tf_child_y_size / 2.0) / tf_child_y_size
+
+            # Our offsets consist of 6 parts:
             # - base coords give offsets to x and y (depend on zoom level)
             # - shift from the parallax effect (depends on mouse pos)
             # - correction to the zoom effect (depends on mouse pos)
             # - offsets from the container (depend on zoom level)
+            # - offsets from the transfrom xpos and ypos (depend on zoom level)
+            # - offsets from the transfrom xanchor and yanchor (depend on zoom level)
             self._transform.xoffset = (
                 self._x*(1.0 + zoom_factor)
                 + available_x_shift*(1.0 - mouse_x_factor)
                 - zoom_factor*self.mouse_x
                 - container_offset_x*(1.0 + zoom_factor)
+                - tf_x_pos * -zoom_factor
+                + tf_x_anchor * -zoom_factor
```

Comment:
I think this makes the math a little clearer. Basically we're taking the distance between the transform's position and its anchor point, then scaling that distance based on the zoom factor.

```suggestion
                + zoom_factor * (tf_x_pos - tf_x_anchor)
```

Reply:
Yeah agree, I have an unpushed commit which did something similar.